datahost-ld-openapi/src/tpximpact/datahost/ldapi/routes/revision.clj

```diff
@@ -62,7 +71,16 @@
 (defn post-revision-changes-route-config [triplestore change-store]
   {:summary "Add changes to a Revision via a CSV file."
    :handler (partial handlers/post-change triplestore change-store)
+   :middleware [[middleware/json-only :json-only]
+                [(partial middleware/flag-resource-exists triplestore
+                          :dh/Revision ::revision) :resource-exists?]
```

Comment:
Generally looks good.

I am currently thinking: If we are already checkng that the Revision resource exists in the middleware, then you could skip the duplicate and now redundant check on the `handlers/post-change` `cond` logic.

You could make that an `if` form, putting the "happy path" `db/insert-change` block where the "not found" `:else` position is currently located. It will be cleaner simpler code and there won't be an additional unessessary hit on the DB.

https://github.com/Swirrl/datahost-prototypes/blob/e539ab6ad8ce25bbd6f78f98fa1216175ad0f6bf/datahost-ld-openapi/src/tpximpact/datahost/ldapi/handlers.clj#L260-L281

There could be more handlers that could be simplified in this way, with redundant code removed.

Reply:
Yes, that's my thinking - there's a `resource-exist?` middleware in the new `middleware` namespace that would short-circuit with 404 when resource does not exist. I didn't add it to the existing routes to prevent this PR from growing.